tools/x-repo-types/xrt_tmpl.go.tmpl

```diff
@@ -0,0 +1,526 @@
+// DO NOT MODIFY!!!! Template generated by 'make template4xrt'.
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"regexp"
+	"sort"
+	"strings"
+	"unicode"
+
+	xpkg "{{.XModulePath}}/{{.XPackagePath}}"  					/* TEMPLATE ONLY  */
+	ypkg "{{.YModulePath}}/{{.YPackagePath}}"  					/* TEMPLATE ONLY  */
+	// 	// ypkg "github.com/algorand/go-algorand-sdk/v2/types"     // 	/* GENERATOR ONLY */
+	// 	xpkg "github.com/algorand/go-algorand/data/bookkeeping"  // 	/* GENERATOR ONLY */
+	// 	ypkg "github.com/algorand/go-algorand/ledger/ledgercore" // 	/* GENERATOR ONLY */  <--- this one is just for getting the build to pass
+)
+
+// diffExclusions is a set of types to exclude. The string keys are assumed to be generated by
+// the `Type.String()` method, and these values are ignored when building up the diff.
+// These exclusions represent types that are known to serialize in the same way, but are
+// would not do so without special treatment.
+// For example, `MicrAlgos` are a struct in go-algorand but a uint64 in go-algorand-sdk.
+var diffExclusions = map[string]bool{
+	`github.com/algorand/go-algorand/data/basics :: "basics.MicroAlgos" (struct)`: true,
+}
+
+// Main is not main for compilation purposes. (see `make template4xrt` for why it's still Main)
+func main() { // replaced by main() in `make template4xrt`
```

Comment:
nit: consider a non-template go file which accepts the two Type objects so that the template doesn't need to be so big.

Reply:
I see what you did with `make template4xrt`. If the template was simple, you could skip that part.

Reply:
I experimented a bit trying to only keep the essential parts in the template and carving out all the type analysis into a separate `typeAnalysis.go` file. However, it looks like I would also need to send that file into the temp directory as the program fails when trying to run the template.

But I'm open to further discussion.

Reply:
Ok, after a bit more fiddling, I was able to slim down the template: 0dd2eeebf9490d8f8266d1ab0a32480ca2c8571b